Obsidian/WorldData/BlockUpdates.cs

```diff
@@ -10,170 +10,183 @@ internal static class BlockUpdates
         internal static async Task<bool> HandleFallingBlock(BlockUpdate blockUpdate)
         {
             if (blockUpdate.block is null) { return false; }
+
             var world = blockUpdate.world;
             var location = blockUpdate.position;
             var material = blockUpdate.block.Value.Material;
-            if (world.GetBlock(location + Vector.Down) is Block below && (Block.Replaceable.Contains(below.Material) || below.IsFluid))
+            if (world.GetBlock(location + Vector.Down) is Block below &&
+                (Block.Replaceable.Contains(below.Material) || below.IsFluid))
             {
                 world.SetBlock(location, Block.Air);
                 world.SpawnFallingBlock(location, material);
                 return true;
             }
+
             return false;
         }
 
         internal static async Task<bool> HandleLiquidPhysics(BlockUpdate blockUpdate)
         {
-            if (blockUpdate.block is null) { return false; }
-            var block = blockUpdate.block.Value;
-            var world = blockUpdate.world;
-            var location = blockUpdate.position;
-            int state = block.State;
-            Vector belowPos = location + Vector.Down;
-            
-            // Handle the initial search for closet path downwards.
-            // Just going to do a crappy pathfind for now. We can do
-            // proper pathfinding some other time.
-            if (state == 0)
+            return await Task.Run(() =>
```

Comment:
`await` may be left out (and then method's `async`).

Reply:
Unsure what you mean here? Do you want me to make the method sync? As it'll break the code which is calling it with an await.

Reply:
I don't see why it should break; it would synchronously return a task, which would then be asynchronously awaited from some asynchronous method. That seems completely valid to me, but correct me if I'm wrong.

Reply:
Oh right, that's what it is doing before, so either we have the async declaration on the method or we do `Task.FromResult(false)`, is returning Task.FromResult okay? @Seb-stian

Reply:
Yes, that would be fine.

Reply:
Changed in commit